README.md

```diff
@@ -159,39 +160,52 @@ In the case of an anonymous [optional parameter](#optional-parameters) the type
 toggle(String, ?: Boolean) => Boolean
 ```
 
-### Constructors
-
-Constructors in JavaScript require the `new` keyword. You can identify a constructor signature using the `new` keyword as if you were demonstrating usage:
+### Builtin Types
 
 ```js
-new User({ username: String }) => UserInstance
+Any, Array, Boolean, Function, Number, Object, String, Symbol, Void
 ```
 
-In JavaScript, a class or constructor is not synonymous with an interface. The class or constructor definition describe the function signature to create the object instances. A separate signature is needed to describe the instances created by the function. For that, use a separate interface with a different name:
+Many builtin types are named after JavaScript constructors. Many syntax highlighters will make the types stand out when the signature is rendered in the docs.
+These are reserved types; hence they can't be overshadowed.
+
+#### The `Any` Type
+
+The special type `Any` means that any type is allowed:
 
 ```js
-interface UserInstance {
-  username: String,
-  credentials: String
-}
+(...args: Any[]) => Array
 ```
 
-### Array Types
+#### The `Void` Type
 
-Arrays with typed contents can be represented like this:
+The special type `Void` should only be used to indicate that a function returns nothing (i.e., `undefined`).
 
 ```js
-Number[]
+set(name: String, value: String) => Void
 ```
 
-### Union Types
+#### The `Predicate` Type
 
-Union types are denoted with the pipe symbol, `|`:
+The special type `Predicate` is a function with the following signature:
 
 ```js
-(userInput: String|Number) => String|Number
+(...args: Any[]) => Boolean
 ```
 
+#### The Object Types
+
+We offer a list of extra object types for convenience:
+- Set
+- Map
+- Date
+- Error
+- Promise
+- RegExp
+- Proxy
+
+Unlike reserved [builtin types](#builtin-types), you can redeclare them.
```

Comment:
> I gave you an example for custom errors

This issue is easily sidestepped by providing custom names for custom errors, i.e., `NotFoundError`.

> Let's say the user wants all his maps to be WeakMap he will just redeclare Map and he's done.

Except now the users of the API may be expecting regular `Map` and instead `WeakMap` is expected, which can lead to confusion, which is exactly why I think this is a bad idea.

> I just realized that RegExp should be a reserved builtin type since it has a literal notation as well.

This I agree with, but not simply because it has a literal notation.